Source/JavaScriptCore/runtime/IteratorOperations.h

```diff
@@ -214,7 +265,22 @@ void forEachInIteratorProtocol(JSGlobalObject* globalObject, JSValue iterable, N
     auto& vm = getVM(globalObject);
     auto scope = DECLARE_THROW_SCOPE(vm);
 
-    if (auto* setIterator = jsDynamicCast<JSSetIterator*>(iterable)) {
+    if (auto* mapIterator = jsDynamicCast<JSMapIterator*>(iterable)) {
+        if (mapIteratorProtocolIsFastAndNonObservable(vm, mapIterator)) {
+            if (JSMap* iteratedMap = jsCast<JSMap*>(mapIterator->iteratedObject())) {
+                if (iteratedMap->isIteratorProtocolFastAndNonObservable()) {
```

Comment:
Why do you need both of these checks? Once the iterator says it's not observable you shouldn't need to check the map again? I suppose ditto for the Set case below too.

Reply:
You're right! I removed the second check